test/test_dists.jl

```diff
@@ -405,6 +406,19 @@ end
             @test mahalanobis(x, y, Q) == sqrt(sqmahalanobis(x, y, Q))
             @test eltype(mahalanobis(x, y, Q)) == T
         end
+        A = rand(T, length(x), length(x))
+        S = A + A' - I
+        @test_throws DomainError SqMahalanobis(A)
```

Comment:
This test requires setting a seed. Maybe it's a bit too specific then? Also, the error is not thrown by Distances.jl, but by `eigmin` in `LinearAlgebra`.

Reply:
You could hardcode a matrix that reproduces this? Otherwise we should use StableRNGs ideally.

Reply:
How about removing it for now, and add a `@test_throws` when we start to throw concrete errors from within `Distances.jl`? When we start to do that, we will hardly throw a `DomainError`, but an `ArgumentError`...

Reply:
Why not, but you still want to check that the logs contain the warning, right?

Reply:
Yes!